app/cdap/components/WranglerGrid/SelectColumnPanel/ColumnsList/index.tsx

```diff
@@ -0,0 +1,218 @@
+import SearchIcon from '@material-ui/icons/Search';
+import styled from 'styled-components';
+import { grey } from '@material-ui/core/colors';
+import {
+  getColumnsSupportedType,
+  getFilteredColumn,
+} from 'components/WranglerGrid/SelectColumnPanel/utils';
+
+const SearchIconComponent = styled(SearchIcon)`
+  &.MuiSvgIcon-root {
+    width: 24px;
+    height: 24px;
+  }
+`;
+
+const SelectColumnSearchInput = styled.input`
+  margin-right: 5px;
+  border: none;
+  border-bottom: 1px solid transparent;
+  margin-bottom: 5px;
+  &:focus {
+    border-bottom: 1px solid ${grey[700]};
+    outline: none;
+  }
+`;
+
+const SelectColumnWrapper = styled(Box)`
+  height: 90%;
+`;
+
+const SelectColumnInnerWrapper = styled(Box)`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-top: 10px;
+`;
+
+const FlexWrapper = styled(Box)`
+  display: flex;
+  height: 100%;
+  align-items: center;
+`;
+
+const CenterAlignBox = styled(Box)`
+  text-align: center;
+`;
+
+const SelectColumnSearchBox = styled(Box)`
+  position: relative;
+  display: flex;
+`;
+
+const SearchIconButton = styled(IconButton)`
+  padding: 5px 0px 5px 5px;
+
+  &.MuiIconButton-root:hover {
+    background-color: transparent;
+  }
+  & .MuiTouchRipple-root {
+    display: none;
+  }
+`;
+
+export default function({
+  transformationDataType,
+  selectedColumnsCount,
+  columnsList,
+  setSelectedColumns,
+  dataQuality,
+  transformationName,
+  selectedColumns,
+}: ISelectColumnsListProps) {
+  const [columns, setColumns] = useState<IHeaderNamesList[]>(columnsList);
+  const [isSingleSelection, setIsSingleSelection] = useState<boolean>(true);
+  const ref = useRef(null);
+
+  const columnsAsPerType = getColumnsSupportedType(transformationDataType, columnsList);
+  const filteredColumnsOnType = getFilteredColumn(transformationDataType, columnsList);
+
+  useEffect(() => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN?.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName
+    );
+
+    if (multiSelect.length) {
+      setIsSingleSelection(false);
+    }
+    setColumns(filteredColumnsOnType);
+  }, []);
+
+  const onSingleSelection = (column: IHeaderNamesList) => {
+    setSelectedColumns([column]);
+  };
+
+  const onMultipleSelection = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    column: IHeaderNamesList
+  ) => {
+    if (event.target.checked) {
+      setSelectedColumns((prev) => [...prev, column]);
+    } else {
+      const indexOfUnchecked = selectedColumns.findIndex(
+        (columnDetail) => columnDetail.label === column.label
+      );
+      if (indexOfUnchecked > -1) {
+        setSelectedColumns(() => selectedColumns.filter((_, index) => index !== indexOfUnchecked));
+      }
+    }
+  };
+
+  const handleDisableCheckbox = () => {
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN.filter(
```

Comment:
Again here this MULTI_SELECTION_COLUMN should be an object instead of this weird filter of an unchanging array. Also why can't you use the isSingleSelection value below instead of calculating this again?

Reply:
Replaced filter with findIndex.
In this isSingleSelection value can't be used as we are having another check  which specifies whether more than two columns can be selected